frontend/model/contracts/group.js

```diff
@@ -348,24 +348,21 @@ const leaveChatRoomAction = async (groupID, state, chatRoomID, memberID, actorID
     data: sendingData,
     ...extraParams
   }).then(() => {
-    // using 'state/vuex/state' here instead of 'chelonia/rootState' to fetch
-    // the identityContractID because although both 'chelonia/rootState' and
-    // 'state/vuex/state' both have the same logged in information
-    // all calls to 'state/vuex/state' will need to be replaced in the future
-    // with something else. Using 'state/vuex/state' here now makes it easier
-    // to find these calls in the future.
-    // `chelonia/rootState` contains that same information but it's a bad
-    // choice because:
-    //    1. Once sandboxing is implemented, it may need to be an async call
-    //    2. Generally, contracts should _not_ access the root state because
-    //       it makes it difficult or impossible to contain them (meaning there
-    //       would be no point in sandboxing)
-    // Instead, in the future contracts will have an 'environment', provided
-    // by Chelonia, which will include global / environment / ambient
-    // information they need.
-    if (memberID === sbp('state/vuex/state').loggedIn.identityContractID) {
-      sbp('okTurtles.events/emit', LEFT_CHATROOM, { identityContractID: memberID, groupContractID: groupID, chatRoomID })
-    }
+    // NOTE: using 'state/vuex/state' instead of 'chelonia/rootState' to fetch
+    //       the identityContractID because although both 'chelonia/rootState' and
+    //       'state/vuex/state' both have the same logged in information
+    //       all calls to 'state/vuex/state' will need to be replaced in the future
+    //       with something else. Using 'state/vuex/state' here now makes it easier
+    //       to find these calls in the future.
+    //       `chelonia/rootState` contains that same information but it's a bad
+    //       choice because:
+    //          1. Once sandboxing is implemented, it may need to be an async call
+    //          2. Generally, contracts should _not_ access the root state because
+    //             it makes it difficult or impossible to contain them (meaning there
+    //             would be no point in sandboxing)
+    //       Instead, in the future contracts will have an 'environment', provided
+    //       by Chelonia, which will include global / environment / ambient
+    //       information they need.
```

Comment:
There are multiple problems here.

1. The code for which this was a comment was moved. Now this is a `.then` handler that does nothing and the comment here is not for anything.
2. The code below with `.catch` is also nonsensical.

So to fix this, please move this comment to where you moved `sbp('state/vuex/state').loggedIn.identityContractID`, because it must stay with it. In this case, above line 1265.

Then remove the `.then` that does nothing, and replace the two `.catch` with one:

```js
  }).catch((e) => {
    if (
      leavingGroup &&
      (e?.name === 'ChelErrorSignatureKeyNotFound' || (
        e?.name === 'GIErrorUIRuntimeError' &&
        (
          ['ChelErrorSignatureKeyNotFound', 'GIErrorMissingSigningKeyError'].includes(e?.cause?.name) ||
          e?.cause?.name === 'GIChatroomNotMemberError'
        )
      ))
    ) {
      // This is fine; it just means we were removed by someone else
      return
    }
    console.warn('[gi.contracts/group] Error sending chatroom leave action', e)
  })
```